python/datafusion/dataframe.py

```diff
@@ -175,7 +178,23 @@ def with_column_renamed(self, old_name: str, new_name: str) -> DataFrame:
         Returns:
             DataFrame with the column renamed.
         """
-        return DataFrame(self.df.with_column_renamed(old_name, new_name))
+        return DataFrame(self.df.rename({old_name: new_name}))
+
+    def rename(self, mapping: dict[str, str]) -> DataFrame:
+        r"""Rename one or multiple columns by applying a new projection.
+
+        This is a no-op if the column to be renamed does not exist.
+
+        The method supports case sensitive rename with wrapping column name
+        into one the following symbols (" or ' or \`).
+
+        Args:
+            mapping (dict[str, str]): mapping of old (key) to new (value) names
```

Comment:
I don't think we need to duplicate the type hints both in the args list and in the function declaration